homeassistant/components/radiotherm/climate.py

```diff
@@ -264,7 +265,22 @@ def update(self):
         # First time - get the name from the thermostat.  This is
         # normally set in the radio thermostat web app.
         if self._name is None:
-            self._name = self.device.name["raw"]
+            try:
+                self._name = self.device.name["raw"]
+            except radiotherm.validate.RadiothermTstatError:
+                _LOGGER.warning(
+                    "%s (%s) was busy (invalid value returned)",
+                    self._name,
+                    self.device.host,
+                )
+                return
+            except timeout:
+                _LOGGER.warning(
+                    "Timeout waiting for response from %s (%s)",
+                    self._name,
+                    self.device.host,
+                )
+                return
```

Comment:
This code is duplicated quite a bit.  Can we do all the device communication in one try block instead?

Reply:
Fair call; I was originally trying to minimize changes, but this cleanup certainly seems innocuous. I confirmed with local testing that everything still works.